### ISSUE TITLE
auro-interruption: Screen Reader Accessibility

### ISSUE THREAD

**Issue by aesthetiques** [AlaskaAirlines/auro-interruption#85]
## To Reproduce
1. Go to https://auro.alaskaair.com/components/auro/interruption/dialog
2. Use a screen reader to navigate to and select any dialog trigger
3. When it opens, the screen reader will give instructions for accessing the dialog. For VoiceOver, this is VO (default ctrl + opt) + shift + down arrow.  This will not work until you first do VO + shift + up arrow, and then repeat.

## Expected behavior
When accessing the auro-dialog, it should behave as the screen reader expects, and immediately enter the context with VO + shift + down arrow.

## Screenshots
Once activated, cannot access with normal navigation binds.
![image](https://user-images.githubusercontent.com/25255197/188243280-838fbb07-84f6-4181-b95f-27b54dfe1595.png)

## Desktop (please complete the following information):
 - OS: macOS Big Sur
 - Browser Chrome/Firefox
 - Version 104, 103 respectively

**Comment by blackfalcon** [comment]
This issue was discussed with @aesthetiques and the feedback was, that while this is a little cumbersome, this is not breaking any accessibility rules that we are aware of. 

This comment is not disregarding the opportunity for a better user experience.

**Comment by jesussilverioea** [comment]
Could we get an update on this issue?
Thank you 😁

**Comment by blackfalcon** [comment]
Moving this back to `refinement` so that it is discussed with the team at the next opportunity.

**Comment by blackfalcon** [comment]
In further review of this issue, we feel that the problem has more to do with our solution of using `inert` and `aria-hidden="true"` is not really providing the best solution for tab key trapping within the dialog element. 

We will need to review this further to discuss options around re-developing the tab-trap solution and not using `intert`. 

Meeting has been scheduled.

**Comment by blackfalcon** [comment]
Comments from the discussion: 

1. @jason-capsule42  and @jordanjones243 offered an option where the first tabbable element is picked up within the scope of another element and applying a focus trap 
2. There is a demo of another option that is more modern and has less code than previous W3C examples https://allyjs.io/tutorials/accessible-dialog.html#interactive-demo

In short, the work suggestion here is to remove the `inert` feature and go to a less invasive solution to make the surrounding DOM less fragile. 

### Research

What are the best practices for escaping the tab-trap? E.g. is it correct to use `esc` or another method other than an action like `X` to close the dialog experience? 

### Scope

Review the previous suggestions and provide estimates at sizing/refinement 

### Exit criteria 

The refactoring of this feature should provide the same experience for tab-trap but with better results. E.g. the outer element is NOT part of the tab index, also the tabs will NOT go outside the scope of the window and into the browser.

**Comment by braven112** [comment]
When the interruption is opened we need to focus on the dialog or drawer without changing the focus styling of the element.

Make sure focus is trapped in interruption while it is open until a user actively closes it.

**Comment by jordanjones243** [comment]
After reviewing many examples of modal dialogs and the accessibility requirements, I have found that modal dialogs behave in two different ways when accessed by VO/Screenreaders:
- The dialog opens and it the VO command tells you to press: Control + Option + Shift + Up Arrow (what we see from this ticket)
- The dialog opens and it immediately gives focus to the first focusable element, where the user is able to then tab through

I was not able to find a dialog that did not follow these two scenarios or exemplify what we expect from the Voiceover commands. I think from here, the options are to either leave as is, or force focus to the first focusable element once the dialog opens, which would then skip the VO command prompting the user to press "Control + Option + Shift + Up Arrow".

**Comment by jordanjones243** [comment]
Here are some of the example dialogs I found:
- https://getbootstrap.com/docs/4.0/components/modal/
- https://www.w3schools.com/bootstrap/bootstrap_modal.asp
- https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog

The team has discussed this issue and has further decided to put it on hold. This issue is not tied to Auro content/code, it has to do with a lack of support for dialogs on screenreaders and browsers.
